File: Programmers/Lv2/다리를 지나는 트럭.py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    res = 0

    bridge = deque([0] * bridge_length)
    truck_weights = deque(truck_weights)

    now = 0

    while len(truck_weights) > 0:
        res += 1
        now -= bridge.popleft()

        if now + truck_weights[0] <= weight:
            now += truck_weights[0]
            bridge.append(truck_weights.popleft())
        else:
            bridge.append(0)

    res += bridge_length
    return res

File: Programmers/Lv2/다리를 지나는 트럭.js
function solution(bridge_length, weight, truck_weights) {
  var answer = 0;

  var bridge = Array(bridge_length).fill(0);
  var now = 0;
  while (truck_weights.length > 0) {
    answer++;
    now -= bridge.shift();

    if (truck_weights[0] + now <= weight) {
      let value = truck_weights.shift();
      now += value;
      bridge.push(value);
    } else {
      bridge.push(0);
    }
  }

  answer += bridge_length;
  return answer;
}
